src/SFA.DAS.Admin.Aan.Web.UnitTests/Controllers/ManageEvent/NetworkEventOrganiserNameControllerTests.cs

```diff
@@ -0,0 +1,95 @@
+        var result = (ViewResult)sut.Get();
+
+        Assert.That(result.Model, Is.TypeOf<EventOrganiserNameViewModel>());
+        var vm = result.Model as EventOrganiserNameViewModel;
+        vm!.CancelLink.Should().Be(AllNetworksUrl);
+        vm.PageTitle.Should().Be(CreateEvent.PageTitle);
+    }
+
+    [Test, MoqAutoData]
+    public void Details_ReturnsExpectedPostLink(
+        [Greedy] NetworkEventOrganiserNameController sut)
+    {
+        sut.AddUrlHelperMock().AddUrlForRoute(RouteNames.ManageEvent.EventOrganiserName, PostUrl);
+        var result = (ViewResult)sut.Get();
+
+        Assert.That(result.Model, Is.TypeOf<EventOrganiserNameViewModel>());
+        var vm = result.Model as EventOrganiserNameViewModel;
+        vm!.PostLink.Should().Be(PostUrl);
+    }
+
+    [TestCase("location 1", null)]
+    [TestCase("location 2", "event online link")]
+    [TestCase(null, "event online link")]
+    public void Post_SetEventDetailsOnSessionModel(string organiserName, string? organiserEmail)
+    {
+        var sessionServiceMock = new Mock<ISessionService>();
+        var validatorMock = new Mock<IValidator<EventOrganiserNameViewModel>>();
+
+        var sessionModel = new EventSessionModel();
+
+        sessionServiceMock.Setup(s => s.Get<EventSessionModel>()).Returns(sessionModel);
+
+        var submitModel = new EventOrganiserNameViewModel { OrganiserName = organiserName, OrganiserEmail = organiserEmail };
+
+        var validationResult = new ValidationResult();
+        validatorMock.Setup(v => v.Validate(submitModel)).Returns(validationResult);
+
+        var sut = new NetworkEventOrganiserNameController(sessionServiceMock.Object, validatorMock.Object);
+
+        sut.AddUrlHelperMock().AddUrlForRoute(RouteNames.NetworkEvents, AllNetworksUrl);
+
+        var result = (RedirectToRouteResult)sut.Post(submitModel);
+
+        sut.ModelState.IsValid.Should().BeTrue();
+        sessionServiceMock.Verify(s => s.Set(It.Is<EventSessionModel>(m => m.OrganiserName == organiserName && m.OrganiserEmail == organiserEmail)));
+        result.RouteName.Should().Be(RouteNames.ManageEvent.EventOrganiserName);
+    }
+
+    [Test, MoqAutoData]
+    public void Post_WhenNoSelectionOfEventLocation_Errors(
+        [Frozen] Mock<ISessionService> sessionServiceMock,
+        [Greedy] NetworkEventOrganiserNameController sut)
+    {
+        sut.AddUrlHelperMock().AddUrlForRoute(RouteNames.NetworkEvents, AllNetworksUrl);
+
+        var sessionModel = new EventSessionModel();
+
+        sessionServiceMock.Setup(s => s.Get<EventSessionModel>()).Returns(sessionModel);
+
+        sut.ModelState.AddModelError("key", "message");
+
+        var submitModel = new EventOrganiserNameViewModel();
+        var result = (ViewResult)sut.Post(submitModel);
+
+        sut.ModelState.IsValid.Should().BeFalse();
+        Assert.That(result.Model, Is.TypeOf<EventOrganiserNameViewModel>());
```

Comment:
Verify SessionService.Set is not invoked.

Reply:
added